File: Programmers/level1/K번째수/siwony_.cs
using System;

public class Solution {
    public int[] solution(int[] array, int[,] commands) {
        
        int commandsTwoDimensionalLen = commands.GetLength(0);
        int[] answer = new int[commandsTwoDimensionalLen];

        for (int i = 0; i < commandsTwoDimensionalLen; i++){
            int startNum = commands[i, 0];
            int endNum = commands[i, 1];
            int pickNum = commands[i, 2] - 1;
            
            int[] arraySort = new int[endNum - startNum + 1];
            int arraySortIdx = 0;
            for (int j = startNum; j <= endNum; j++){
                arraySort[arraySortIdx++] = array[j - 1];

                if (j == commands[i, 1]){
                    Array.Sort(arraySort, (a, b) => a > b ? 1 : -1);
                    answer[i] = arraySort[pickNum];
                }
            }
        }

        return answer;
    }
}

File: Programmers/level1/K번째수/siwony_.py
def solution(arr, commands):
    answer = []
    for com in commands:
        sort_arr = sorted((arr[com[0] - 1 : com[1]]))
        answer.append(sort_arr[com[2] - 1])
    return answer
####테스트 코드####
print(solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3], [4, 4, 1], [1, 7, 3]]))